collect/collect.go

```diff
@@ -131,6 +135,13 @@ func InitChan(tsdbhost *url.URL, root string, ch chan *opentsdb.DataPoint) error
 		slock.Unlock()
 		return
 	})
+	Set("collect.discarded", Tags, func() (i interface{}) {
+		slock.Lock()
```

Comment:
I don't think you even need to lock if you are using the atomic operations. You could use atomic.LoadInt64 and not need locks at all. I wish we had done all these stats like that from the start.